### PR TITLE
added summary for modules table in users

### PR DESCRIPTION
**Pull request by pavanpulicherla18345**

### CODE REVIEW

File: apps/users/serializers.py [hunk 1]
```diff
@@ -227,3 +227,17 @@ def validate(self, attrs):
...
   
         
+class ModulesOptionsSerializer(serializers.ModelSerializer):
+    module_sections = ModModuleSectionsSerializer(many=True, read_only=True, source='modulesections_set')
+
+    class Meta:
+        model = Modules
+        fields = ['module_id', 'module_name', 'description', 'module_sections']
+
+    def get_modules_summary(modules):
```
Comment (QodeFarm) [inline apps/users/serializers.py:237]: change this name to  get_module_sections

File: apps/users/views.py [hunk 2]
```diff
@@ -75,7 +75,15 @@ class ModulesViewSet(viewsets.ModelViewSet):
     serializer_class = ModulesSerializer
 
     def list(self, request, *args, **kwargs):
-        return list_all_objects(self, request, *args, **kwargs)
+        summary = request.query_params.get('summary', 'false').lower() == 'true'
```
Comment (QodeFarm) [inline apps/users/views.py:78]: change summary to sections

File: apps/users/views.py [hunk 3]
```diff
@@ -75,7 +75,15 @@ class ModulesViewSet(viewsets.ModelViewSet):
...
+        if summary:
```
Comment (QodeFarm) [inline apps/users/views.py:79]: change summary to sections

File: apps/users/views.py [hunk 4]
```diff
@@ -75,7 +75,15 @@ class ModulesViewSet(viewsets.ModelViewSet):
...
+            modules = self.filter_queryset(self.get_queryset())
+            data = ModulesOptionsSerializer.get_modules_summary(modules)
```
Comment (QodeFarm) [inline apps/users/views.py:81]: get module_sections here

**Review by QodeFarm — COMMENTED** [review]
api should be modules/?sections=true

File: apps/users/serializers.py (hunk 1 above)
Comment (pavanpulicherla18345) [reply apps/users/serializers.py:237]: Sorry Thulasi I will change it

File: apps/users/views.py (hunk 2 above)
Comment (pavanpulicherla18345) [reply apps/users/views.py:78]: Okay Thulasi.

File: apps/users/views.py (hunk 3 above)
Comment (pavanpulicherla18345) [reply apps/users/views.py:79]: Okay Thulasi.

File: apps/users/views.py (hunk 4 above)
Comment (pavanpulicherla18345) [reply apps/users/views.py:81]: Okay Thulasi.

**Pull request merged**